SOFTENG310-Team4/SMARTJ
# Feat/create components

## Context


The goal of this pull request is to address the need for modularization by separating the VideoRecording and TextAnswer functionalities into their own individual components. This change was prompted by issue #34 - regarding improving code maintainability and readability by decoupling these features.

Closes #34 

## What Changed?


- Created VideoRecordingComponent.js: Extracted all video recording-related logic from the parent component and moved it into this new component.
- Created TextAnswerComponent.js: Moved all text answer related logic into its own separate component.
- Refactored interview practice page to import and use the new VideoRecordingComponent and TextAnswerComponent, ensuring the functionality remains intact.
- Improved the text answer component by adding a submit button to disable the textbox and pause timer.
- Improved the text answer component by adding a start new answer button to reset the textbox and timer.

## How To Review


- Start by reviewing the two newly created components (VideoRecordingComponent.js and TextAnswerComponent.js).
- Check the refactored parent component to ensure it imports and integrates the new components properly.
- Ensure there are no regressions in functionality related to the countdown, video recording, and text input submission.

## Testing


- Tested the interview practice page to see if the different answer types worked.

## Risks


NA

## Notes
NA
